src/components/scenes/Fio/FioCreateHandleScene.tsx

```diff
@@ -54,9 +70,71 @@ export const FioCreateHandleScene = ({ navigation }: Props) => {
   const handleRegisterPress = async () => {
     // Register button is disabled if wallet creation isn't finished yet
     if (wallet == null) return
+    const fioName = `${fioHandle}${domainStr}`
+
+    // Check if the fio name is already registered
+    let isAvailRes
+    try {
+      const isAvailResJson = await fetch('https://fio.blockpane.com/v1/chain/avail_check', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          fio_name: fioName
+        })
+      }).then(async res => res.json())
+      isAvailRes = asIsRegisteredRes(isAvailResJson)
+    } catch (e: any) {
+      console.error(e)
+      setErrorText(s.strings.fio_register_handle_error)
+      return
+    }
 
-    // TODO: Check if fioHandle is already registered, create the handle.
-    setErrorText('TODO')
+    if (isAvailRes.is_registered === 1) {
+      setErrorText(sprintf(s.strings.fio_register_handle_taken_error_s, fioName))
+    } else {
+      // Handle is available. Register handle.
+      try {
+        const regAddressRes = await fetch('https://reg.fioprotocol.io/public-api/buy-address', {
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            address: fioName,
+            referralCode: 'edgefree',
+            publicKey: (await wallet.getReceiveAddress()).publicAddress,
+            redirectUrl: '',
+            apiToken: '1aNR92i2HC9MMs7ITvHSrcHu2hEoi1fhk2wLmLGDo670gA'
+          })
+        }).then(async res => res.json())
+
+        // Check registration status
+        try {
+          asRegisterSuccessRes(regAddressRes)
+
+          await dispatch(refreshAllFioAddresses())
+          navigation.pop()
+        } catch (e: any) {
+          // Rejected somehow, see if error is readable
+          try {
```

Comment:
unnecessary try/catch. You're already in one.

Reply:
I'm trying the asRegisteredFailedRes here

Reply:
Simplifying the error handling - using fioPlugin otherMethods and throwing
whatever errors are handled in buyAddressRequest.